### PR TITLE
[in_app_purchase] Fix Java lints

### PR DESCRIPTION
**Pull request by stuartmorgan**
Removes lint-baseline.xml, fixing all resulting warnings.

Also makes some fixes that weren't flagged by the command-line lint, but were flagged by Android Studio's linter, to reduce the noise in the UI. This includes all of the conversions of anonymous subclasses to lambdas, which were entirely tool-generated.

Part of https://github.com/flutter/flutter/issues/88011

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java [hunk 1]
```diff
@@ -111,22 +131,22 @@ void onDetachedFromActivity() {
   }
 
   @Override
-  public void onMethodCall(MethodCall call, MethodChannel.Result result) {
+  public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
```
Comment (gmackall) [inline packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java:134]: It looks like some of the unit tests are failing with an IllegalArgumentException on calls to onMethodCall. 

I've had some difficulties getting the these tests to run locally, but given that the only change was to add @NonNull I'm assuming that some of the tests were passing in null somehow?

Comment (gmackall) [reply packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java:134]: Hmm the tests are definitely not passing in null. I've confirmed that the tests succeed on the commit before this one and not on this one, but I'm somewhat at a loss as to what causes this error:
```
Please provide Old SKU purchase information(token/id) or original external transaction id, not both.
java.lang.IllegalArgumentException: Please provide Old SKU purchase information(token/id) or original external transaction id, not both.
	at com.android.billingclient.api.BillingFlowParams$SubscriptionUpdateParams$Builder.build(com.android.billingclient:billing@@5.2.0:7)
	at com.android.billingclient.api.BillingFlowParams$Builder.build(com.android.billingclient:billing@@5.2.0:42)
	at io.flutter.plugins.inapppurchase.MethodCallHandlerImpl.launchBillingFlow(MethodCallHandlerImpl.java:331)
	at io.flutter.plugins.inapppurchase.MethodCallHandlerImpl.onMethodCall(MethodCallHandlerImpl.java:150)
	at io.flutter.plugins.inapppurchase.MethodCallHandlerTest.launchBillingFlow_null_AccountId_do_not_crash(MethodCallHandlerTest.java:292)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	...
```

**Review by gmackall — APPROVED** [review]
> Oops, I forget this hadn't actually been LGTM 😬 auto-submit bot for the win!

Was actually right about to approve, LGTM!
